subworkflows/local/fastq_align_bwa_markduplicates/main.nf

```diff
@@ -6,52 +6,64 @@ include { BWAMEM2_MEM                   } from '../../../modules/nf-core/bwamem2
 include { BAM_SORT_STATS_SAMTOOLS       } from '../../nf-core/bam_sort_stats_samtools/main'
 include { PICARD_ADDORREPLACEREADGROUPS } from '../../../modules/nf-core/picard/addorreplacereadgroups/main'      
 include { PICARD_MARKDUPLICATES         } from '../../../modules/nf-core/picard/markduplicates/main'
+include { SAMTOOLS_MERGE                } from '../../../modules/nf-core/samtools/merge/main'
 
 workflow FASTQ_ALIGN_BWA_MARKDUPLICATES {
     take:
     ch_reads        // channel (mandatory): [ val(meta), [ path(reads) ] ]
     ch_index        // channel (mandatory): [ val(meta2), path(index) ]
     val_sort_bam    // boolean (mandatory): true or false
     ch_fasta        // channel (mandatory) : [ val(meta3), path(fasta) ]
-    ch_fai        // channel (mandatory) : [ val(meta4), path(fai) ]
+    ch_fai          // channel (mandatory) : [ val(meta4), path(fai) ]
 
     main:
     ch_versions = Channel.empty()
 
-    //
-    // Map reads with BWA
-    //
+    // Map reads with BWA - per lane
+    BWAMEM2_MEM ( 
+        ch_reads, 
+        ch_index, 
+        ch_fasta, 
+        val_sort_bam )
```

Comment:
`val_sort_bam` is passed as `true`. Do we need to sort the BAM twice? Since we sort it later in `BAM_SORT_STATS_SAMTOOLS`. Sorting takes a lot of CPU so best to do it only as needed. If we don't need to sort twice, I'd say we remove `val_sort_bam` as an option and just pass `false`.

Reply:
We have to sort them after merging for sure, yet I am not sure how would they behave if we try to merge unsorted bams. we can give it a try.

Reply:
I just went over the documentation https://www.htslib.org/doc/samtools-merge.html and that is true, we dont need to sort them after merging. I will make the necessary changes

Reply:
okay, samtools sort is also removed.